Samples/runtime/simulator-with-overrides/FaultyMeasurementsSimulator.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+using Microsoft.Quantum.Simulation.Simulators;
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Intrinsic;
+
+namespace Microsoft.Quantum.Samples.SimulatorWithOverrides
+{
+    /// <summary>
+    /// A simulator which extends QuantumSimulator and redefines measurement operation 
+    /// to return the correct measurement result with certain probability 
+    /// and the opposite measurement result in the rest of the cases.
+    /// </summary>
+    public class FaultyMeasurementsSimulator : QuantumSimulator
+    {
+        /// <summary>
+        /// The overriding definition for operation M
+        /// </summary>
+        public class M : QSimM
+        {
+            /// <summary>
+            /// The probability with which the measurement result will be flipped.
+            /// </summary>
+            const double flipProbability = 0.1;
+
+            /// <summary>
+            /// Random numbers generator used to decide when to flip the result.
+            /// </summary>
+            private static readonly System.Random rnd = new System.Random();
+
+            /// <summary>
+            /// The X gate operation used to adjust the state to match flipped measurement results.
+            /// </summary>
+            private static IUnitary<Qubit> gateX;
+
+            public M(FaultyMeasurementsSimulator m) : base(m) { }
+
+            /// <summary>
+            /// The actual definition of what the new operation does.
+            /// </summary>
+            public override Func<Qubit, Result> Body {
+                get {
+                    // Get the original M operation to call it and process the results
+                    Func<Qubit, Result> originalMeasurementOperation = base.Body;
```

Comment:
```suggestion
                    var originalMeasurementOperation = base.Body;
```

Reply:
For this sample I'd really like to keep things as explicit as possible, since people will be using it to start their work with, and figuring out the types of the more unusual variables in a very unfamiliar code base is very tricky.

Reply:
I guess my thinking is that the general coding style to use `var` in sample C# code fits well here, too, in that the type of the property is already given at L44. Similarly at L10, the type is already given once with the expression `new FaultyMeasurementSimulator()`, such that using an explicit type is redundant.

That said, I also appreciate the need to make exceptions to general coding guidelines where there's a specific usecase for doing so, such as where there's need to emphasize a particularly unfamiliar type. With respect to L56, for example, I could see expanding the comment and making the local variable type explicit to help readers of this sample understand that the Q# type `Result` maps to a C# type with the same name — that is definitely not obvious at first, given that Q# `Int` maps to C# `long` and that Q# `'T[]` maps to `IQArray<T>`.